edb/schema/objects.py

```diff
@@ -228,23 +263,24 @@ def required(self):
         return True
 
     @property
-    def is_schema_field(self):
+    def is_schema_field(self) -> bool:
         return False
 
-    def __get__(self, instance, owner):
+    def __get__(self, instance: T, owner: Type[T]) -> Optional[Field]:
+        # XXX what is happening in this method?
```

Comment:
When a `<Field.name>` is accessed on a class, we return the `Field` instance, but when it's accessed on an instance, we return `None` to indicate "no value for this field", since, if there was a value, it would have been in the object's __dict__ and we wouldn't get here.

Now, `Field` is supposed to be non-optional (as opposed to `SchemaField`), but `EphemeralCollection` still relies on this.